errors/errors.go

```diff
@@ -26,28 +23,20 @@ type ArgoError interface {
 	Code() string
 	Message() string
 	JSON() []byte
-	StackTrace() errors.StackTrace
-	Format(s fmt.State, verb rune)
 }
 
 // argoerr is the internal implementation of an Argo error which wraps the error from pkg/errors
 type argoerr struct {
 	code    string
 	message string
-	stracer stackTracer
-}
-
-// stackTracer is interface for error types that have a stack trace
-type stackTracer interface {
-	Error() string
-	StackTrace() errors.StackTrace
+	err     error
 }
 
 // New returns an error with the supplied message.
 // New also records the stack trace at the point it was called.
 func New(code string, message string) error {
-	err := errors.New(message)
-	return argoerr{code, message, err.(stackTracer)}
+	err := fmt.Errorf(message)
```

Comment:
should we moving to `Errorf("%w")` and `errors.Is()`

Reply:
@alexec `Errorf("%w")` is used in the `Wrap` function. I didn't find a need to use `errors.Is()` and instead relied on Go's error library